### PR TITLE
Improved schema and converters for bibliographic data

### PR DESCRIPTION
**Pull request by zuphilip**
This is a start for the tasks in #494. The item types are now adjusted and also some fields are moved. We need to add the other fields and work some more here... Currently, add a reference looks like this:

texture-add-reference

### CODE REVIEW

File: src/converter/r2t/ConvertRefs.js [hunk 1]
```diff
@@ -49,11 +46,14 @@ export default class ConvertRef {
...
+            entityId = NewspaperArticleConverter.import(elementCitation, pubMetaDb)
             break;
-          case 'preprint':
-            entityId = PreprintConverter.import(elementCitation, pubMetaDb)
+          case 'magazine':
+            entityId = MagazineArticleConverter.import(elementCitation, pubMetaDb)
             break;
+          case 'clinicaltrial':
+          case 'preprint':
           case 'report':
```
[the remark this answers is not in the document]
Comment (michael) [reply src/converter/r2t/ConvertRefs.js]: Yes, we can remove them here. We don't do checks for attribute values yet, but in the future we should be able to specify which values are allowed an display an error if you use an unsupported pub type.

File: data/kitchen-sink/manuscript.xml [hunk 2]
```diff
@@ -484,22 +484,24 @@
...
+          <article-title>What if superheroes aren’t really the good guys?</article-title>
+          <source>The Guardian</source>
+          <edition>International Edition</edition>
+          <year>2018</year>
+          <month>4</month>
+          <day>27</day>
+          <!--
+          TODO This new field does not work (also it is in JATS)...
+          <part-title>Film</part-title>
+          -->
```
[the remark this answers is not in the document]
Comment (michael) [reply data/kitchen-sink/manuscript.xml]: Is this causing an error, or just missed on import? I will take a look.

[the remark this answers is not in the document]
Comment (michael) [reply data/kitchen-sink/manuscript.xml]: Ok, this was missing in the schema. I've pushed a change to master. https://github.com/substance/texture/commit/0665b5040c4edfd774d9619240e7681f2c9af6b2

[the remark this answers is not in the document]
Comment (zuphilip) [reply data/kitchen-sink/manuscript.xml]: Thank you. Yes, that works now.

File: src/entities/EntitiesPackage.js [hunk 3]
```diff
@@ -18,10 +18,9 @@ export default {
     config.addNode(BibliographicEntry)
     config.addNode(JournalArticle)
     config.addNode(ConferenceProceeding)
-    config.addNode(ClinicalTrial)
-    config.addNode(Preprint)
     config.addNode(DataPublication)
-    config.addNode(Periodical)
+    config.addNode(MagazineArticle)
+    config.addNode(NewspaperArticle)
     config.addNode(Patent)
```
Comment (zuphilip) [inline src/entities/EntitiesPackage.js]: Do we need to increase the version number some lines above?
```js
version: '1.0.0',
```

Comment (michael) [reply src/entities/EntitiesPackage.js]: We can keep the version while in beta, but once Texture 1.0 is out we should start with versioning the schema, yes!

**Comment by zuphilip** [comment]
@michael Would you be okay if I try to simplify the `r2t/EntityConverters.js` such that not each item types has its own function, but rather one mapping for all of them. Most of the fields could be mapped independently of the item type.

**Comment by michael** [comment]
Sure, if one mapping makes for an easier implementation. Thank you for helping with this!

**Comment by zuphilip** [comment]
How can I test the export to JATS?

**Comment by zuphilip** [comment]
Okay, I refactured that conversion from and to JATS. This should simplify any coming changes for the data model.

**Comment by michael** [comment]
Looking very good. Ping me when ready to review!

**Comment by zuphilip** [comment]
Thank you! The changes here should be ready for review and testing (please do some testing). This should include all changes from #494.

What is then still open afterwards is #497, but I guess we also need there some dar changes and maybe some points are for discussion first...

**Comment by michael** [comment]
👍 

I'm just realising that some forms now get pretty long (e.g. Chapter). Wonder if we should somehow mark the primary fields (which should be entered at minimum) and hide the secondary (optional) fields, with the option to expand them.

**Comment by michael** [comment]
Really liking the normalised model now. Still i don't understand all the details here as I'm really not a domain expert. Great to have you help here @zuphilip! Much appreciated!

**Pull request merged**